src/values/ArrayValue.js

```diff
@@ -9,18 +9,88 @@
 
 /* @flow strict-local */
 
-import type { Realm } from "../realm.js";
+import type { Effects, Realm } from "../realm.js";
 import type { PropertyKeyValue, Descriptor, ObjectKind } from "../types.js";
-import { AbstractValue, ObjectValue, StringValue, NumberValue, Value } from "./index.js";
+import {
+  AbstractValue,
+  BoundFunctionValue,
+  ECMAScriptSourceFunctionValue,
+  NumberValue,
+  ObjectValue,
+  StringValue,
+  Value,
+} from "./index.js";
 import { IsAccessorDescriptor, IsPropertyKey, IsArrayIndex } from "../methods/is.js";
-import { Properties, To } from "../singletons.js";
+import { Havoc, Properties, To, Utils } from "../singletons.js";
 import { type OperationDescriptor } from "../utils/generator.js";
 import invariant from "../invariant.js";
 
+type PossibleNestedOptimizedFunctions = [
+  { func: BoundFunctionValue | ECMAScriptSourceFunctionValue, thisValue: Value },
+];
+
+function handleNestedOptimizedFunctions(
```

Comment:
This name is not any better than "doSomeRandomStuffToSomethingOrOther". It would be less bad if this function were nested inside createAbstractArrayValue. Either way, a more descriptive name would help the reader.